### PR TITLE
Update API docs for ParallelInterleaveDataset

### PR DESCRIPTION
**Pull request by frreiss**
This pull request updates the API docs for the `ParallelInterleaveDataset` op. I added documentation for the inputs and arguments, added a reference to the Python API call that creates instances of this dataset, and added a note that there is a newer implementation under the name `ParallelInterleaveDatasetV2`.

### CODE REVIEW

File: tensorflow/core/api_def/base_api/api_def_ParallelInterleaveDataset.pbtxt [hunk 1]
```diff
@@ -1,6 +1,60 @@
...
+dataset returned by `f`.
+END
+  }
+  in_arg {
+    name: "sloppy"
+    description: <<END
+If `True`, return elements as they become available, even if that means returning
+these elements in a different order.
+If `False`, the operation of this dataset will be deterministic, but the
+dataset may block.
```
Comment (jsimsa) [inline tensorflow/core/api_def/base_api/api_def_ParallelInterleaveDataset.pbtxt]: The dataset may "block" even when sloppy is True. Instead say, that sloppy operation may result in better performance in the presence of stragglers (e.g. due to I/O).

File: tensorflow/core/api_def/base_api/api_def_ParallelInterleaveDataset.pbtxt [hunk 2]
```diff
@@ -18,5 +84,11 @@ when loading data from a variable-latency datastores (e.g. HDFS, GCS), as it
 allows the training step to proceed so long as some data is available.
 
 !! WARNING !! This dataset is not deterministic!
```
Comment (jsimsa) [inline tensorflow/core/api_def/base_api/api_def_ParallelInterleaveDataset.pbtxt]: I would update this bit to say the it is not deterministic when sloppy is set to `True`.

**Review by jsimsa — CHANGES_REQUESTED** [review]
Thanks Fred. Left a couple of minor comments.

As a side note, in terms of priority, it would make sense to first update docs for operations that are use by the "core" tf.data Python API.

File: tensorflow/core/api_def/base_api/api_def_ParallelInterleaveDataset.pbtxt [hunk 3]
```diff
@@ -1,6 +1,61 @@
...
+    description: <<END
+Number of elements at a time to produce from each interleaved invocation of a
+dataset returned by `f`.
+END
+  }
+  in_arg {
+    name: "sloppy"
+    description: <<END
+If `True`, return elements as they become available, even if that means returning
+these elements in a different order. Sloppy operation may result in better
```
Comment (jsimsa) [inline tensorflow/core/api_def/base_api/api_def_ParallelInterleaveDataset.pbtxt]: replace "different" with "non-deterministic"

File: tensorflow/core/api_def/base_api/api_def_ParallelInterleaveDataset.pbtxt [hunk 4]
```diff
@@ -1,6 +1,61 @@
...
+performance in the presence of stragglers, but the dataset will still block if
+all of its open streams are blocked.
+If `False`, always return elements in the same order.
```
Comment (jsimsa) [inline tensorflow/core/api_def/base_api/api_def_ParallelInterleaveDataset.pbtxt]: replace "same" with "deterministic"

**Review by jsimsa — COMMENTED** [review]
Looks good to me modulo to small changes.

File: tensorflow/core/api_def/base_api/api_def_ParallelInterleaveDataset.pbtxt (hunk 3 above)
Comment (frreiss) [reply tensorflow/core/api_def/base_api/api_def_ParallelInterleaveDataset.pbtxt]: Fixed in commit 74b0247

File: tensorflow/core/api_def/base_api/api_def_ParallelInterleaveDataset.pbtxt (hunk 4 above)
Comment (frreiss) [reply tensorflow/core/api_def/base_api/api_def_ParallelInterleaveDataset.pbtxt]: Fixed in commit 74b0247

**Review by jsimsa — APPROVED** [review]

**Pull request merged**